vault/config.go

```diff
@@ -326,6 +328,23 @@ func (cl *ConfigLoader) populateFromConfigFile(config *Config, profileName strin
 	if config.STSRegionalEndpoints == "" {
 		config.STSRegionalEndpoints = psection.STSRegionalEndpoints
 	}
+	if sessionTags := psection.SessionTags; sessionTags != "" && len(config.SessionTags) == 0 {
+		if config.SessionTags == nil {
+			config.SessionTags = make(map[string]string)
+		}
+		for _, tag := range strings.Split(sessionTags, ",") {
+			kvPair := strings.SplitN(tag, "=", 2)
+			if len(kvPair) != 2 {
+				return errors.New("Failed to parse session_tags profile setting, must be <key1>=<value1>,[<key2>=<value2>[,...]]")
+			}
+			config.SessionTags[strings.TrimSpace(kvPair[0])] = strings.TrimSpace(kvPair[1])
+		}
+	}
+	if transitiveSessionTags := psection.TransitiveSessionTags; transitiveSessionTags != "" && len(config.TransitiveSessionTags) == 0 {
+		for _, tag := range strings.Split(transitiveSessionTags, ",") {
+			config.TransitiveSessionTags = append(config.TransitiveSessionTags, strings.TrimSpace(tag))
+		}
```

Comment:
could you split this logic out into it's own function?

Reply:
please check the latest change, I created a couple of setters for `vault.Config` that parse string into the corresponding map and slice.

I'm not 100% confident that it's the best approach - it's "economical", but kind of moves parsing away from `vault.ProfileSection` to `vault.Config`. Please tell me your thoughts on that.